Emails: Update copy and hide placeholders on Email Comparison page

stephanethomas opened the pull request:
This pull requests updates the `Email Comparison` page to:

* Update the description of Google Workspace to match changes from [this pull request](https://github.com/Automattic/wp-calypso/pull/58186)
* Refer to mailboxes instead of users in the price (i.e. `/user /month billed annually`)
* Remove duplicate billing information for Google Workspace (i.e. `billed annually`)
* Remove placeholders that just duplicate the labels of each form field

Note this last change is also visible on the `Add New Mailboxes` and `Set Up Mailbox` pages since they use the same forms for Titan and Google Workspace.

##### Email Comparison

Before | After
------ | -----
![image](https://user-images.githubusercontent.com/594356/142262583-afc2ed49-592a-4936-86ed-53be398765d3.png) <br/><br/><br/><br/><br/><br/><br/><br/>| ![image](https://user-images.githubusercontent.com/594356/142262644-fde0baf1-3ac5-4a96-8e6e-c72e4a2ae080.png)


##### Add New Mailboxes (Titan)

Before | After
------ | -----
![image](https://user-images.githubusercontent.com/594356/142262401-11370264-2084-4535-a230-d4aab0fbe2f5.png) | ![image](https://user-images.githubusercontent.com/594356/142262468-6ab5b121-8090-4f98-baa8-30c827941d44.png)


##### Add New Mailboxes (Google Workspace)

Before | After
------ | -----
![image](https://user-images.githubusercontent.com/594356/142262182-fc452653-1d4b-4e5c-8a4e-143a8be88524.png) | ![image](https://user-images.githubusercontent.com/594356/142262238-0ad4153d-8741-4cd2-ab62-0ed21d77fe74.png)

##### Set Up Mailbox

Before | After
------ | -----
![image](https://user-images.githubusercontent.com/594356/142262036-3ca2dad1-374b-46dc-a1cd-a6e312a72d9e.png) | ![image](https://user-images.githubusercontent.com/594356/142262075-35c14765-a3cb-4c6a-9563-aca15af41d5c.png)


#### Testing instructions

1. Run `git checkout update/email-comparison-page` and start your server, or access a [live branch](https://github.com/Automattic/wp-calypso/pull/58193#issuecomment-971861463)
2. Log into a WordPress.com account that has Titan, Google Workspace, and a domain with no email
3. Open the [`Emails` page](http://calypso.localhost:3000/email)
4. Click the `Add Email` button to access the `Email Comparison` page
5. Assert that you see the changes mentioned above
6. Go back to the `Emails` page
7. Click a Titan account
8. Click the `Set up mailbox` button to access the `Set Up Mailbox` page
9. Assert that placeholders are no longer displayed
10. Go back, and click the `Add new mailboxes` option to access the `Add New Mailboxes` page
11. Assert that placeholders are no longer displayed
12. Go back to the `Emails` page
13. Click a Google Workspace account
14. Click the `Add new mailboxes` option to access the `Add New Mailboxes` page
15. Assert that placeholders are no longer displayed

stephanethomas committed:
G Suite: Update Google Workspace description on Email Comparison page

stephanethomas committed:
Emails: Update Email Comparison page to refer to mailboxes instead of users

stephanethomas committed:
G Suite: Remove duplicate billing information for Google Workspace on Email Comparison page

stephanethomas committed:
G Suite: Remove placeholders from new Google Workspace mailboxes form

stephanethomas committed:
Titan: Remove placeholders from new Titan mailboxes form

a8ci18n commented on client/my-sites/email/email-providers-comparison/index.jsx:
```diff
@@ -357,7 +357,7 @@ class EmailProvidersComparison extends Component {
 		const monthlyPrice = getMonthlyPrice( gSuiteProduct?.cost ?? null, currencyCode );
 		const formattedPrice = productIsDiscounted
-			? translate( '{{fullPrice/}} {{discountedPrice/}} /user /month billed annually', {
+			? translate( '{{fullPrice/}} {{discountedPrice/}} /mailbox /month billed annually', {
```
:ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

a8ci18n commented on client/my-sites/email/email-providers-comparison/index.jsx:
```diff
@@ -369,7 +369,7 @@ class EmailProvidersComparison extends Component {
 						'{{fullPrice/}} is the formatted full price, e.g. $20; {{discountedPrice/}} is the discounted, formatted price, e.g. $10',
 			  } )
-			: translate( '{{price/}} /user /month billed annually', {
+			: translate( '{{price/}} /mailbox /month billed annually', {
```
:ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

a8ci18n commented on client/my-sites/email/email-providers-comparison/index.jsx:
```diff
@@ -499,7 +489,7 @@ class EmailProvidersComparison extends Component {
 		} = this.props;
 
-		const formattedPrice = translate( '{{price/}} /user /month billed monthly', {
+		const formattedPrice = translate( '{{price/}} /mailbox /month billed monthly', {
```
:ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6979612

Thank you @stephanethomas for including a screenshot in the description! This is really helpful for our translators.

daledupreez commented:
> Expand the Google Workspace form by default so users can compare our two main solutions easily

This is a feature of the current design/UX that was a core part of the design. I think we should defer removing that as part of this PR, as I think it needs some dedicated discussion before we make a call on this front.

olaseni reviewed (dismissed):
~~LGTM! All screens are verified~~
Is there a reason why we are expanding two forms?

stephanethomas commented:
> Expand the Google Workspace form by default so users can compare our two main solutions easily

I've rebased the branch, and removed the commit responsible for that change 👍.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6979612).
